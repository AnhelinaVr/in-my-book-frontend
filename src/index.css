@import '@fontsource/montserrat';

body {
  margin: 0;
  font-family: "Montserrat", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}
