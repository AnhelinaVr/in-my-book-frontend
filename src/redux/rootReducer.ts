import alertReducer from './alert/slice';
import { combineReducers } from '@reduxjs/toolkit';

const rootReducer = combineReducers({
  alert: alertReducer,
});

export type AppState = ReturnType<typeof rootReducer>;

export default rootReducer;
