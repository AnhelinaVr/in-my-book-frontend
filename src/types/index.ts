export type Link = {
  name: string;
  path: string;
};

export enum AlertType {
  ERROR = 'error',
  SUCCESS = 'success',
}
