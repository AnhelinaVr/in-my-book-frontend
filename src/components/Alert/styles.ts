import { AlertType } from '../../types';
import { SxProps } from '@mui/system';
import { Theme } from '@mui/material';

export const getStyles = (theme: Theme): Record<string, SxProps> => ({
  root: {
    fontSize: theme.typography.fontSize,
    width: '100%',
    marginTop: theme.spacing(22.5),
    boxShadow: `${theme.spacing(0, 0, 0.5)}, gray`, // TODO: get color from palette
  },
  [AlertType.SUCCESS]: {
    background: theme.palette.success.main,
  },
  [AlertType.ERROR]: {
    background: theme.palette.error.main,
  },
});
