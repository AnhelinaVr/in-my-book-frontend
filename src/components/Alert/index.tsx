import { Alert as MuiAlert, Snackbar } from '@mui/material';
import { Theme, useTheme } from '@mui/material';
import { useDispatch, useSelector } from 'react-redux';
import { AlertType } from '../../types';
import { AppState } from '../../redux/rootReducer';
import ErrorIcon from '@mui/icons-material/WarningRounded';
import SuccessIcon from '@mui/icons-material/CheckRounded';
import { getStyles } from './styles';
import { hideAlert } from '../../redux/alert/actions';

const getIcon = (type: AlertType, theme: Theme) =>
  type === AlertType.SUCCESS ? (
    <SuccessIcon sx={{ color: theme.palette.success.dark }} />
  ) : (
    <ErrorIcon sx={{ color: theme.palette.warning.main }} />
  );

export const Alert = () => {
  const dispatch = useDispatch();
  const theme = useTheme();
  const styles = getStyles(theme);

  const alert = useSelector((state: AppState) => state.alert);

  const handleClose = (
    _event?: Event | React.SyntheticEvent,
    reason?: string,
  ) => {
    if (reason === 'clickaway') {
      return;
    }

    dispatch(hideAlert());
  };

  return (
    <>
      {alert && (
        <Snackbar
          open={alert.isOpen}
          anchorOrigin={{ vertical: 'top', horizontal: 'center' }}
          autoHideDuration={alert.timeout}
          onClose={handleClose}
        >
          <MuiAlert
            severity={alert.type}
            onClose={handleClose}
            sx={() => ({ ...styles.root, ...styles[alert.type] })}
            icon={getIcon(alert.type, theme)}
          >
            {alert.message}
          </MuiAlert>
        </Snackbar>
      )}
    </>
  );
};
