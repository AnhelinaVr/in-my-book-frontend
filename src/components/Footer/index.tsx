import { Box, Link, Typography, useTheme } from '@mui/material';
import { ReactComponent as GithubIcon } from './images/gh-icon.svg';
import { getStyles } from './styles';

export const Footer = () => {
  const theme = useTheme();
  const styles = getStyles(theme);

  return (
    <Box sx={styles.container}>
      <Box sx={styles.github}>
        <GithubIcon />
        <Link
          href="https://github.com/in-my-book-team"
          target="_blank"
          sx={styles.githubLink}
        >
          GitHub
        </Link>
      </Box>
      <Typography
        sx={{ fontSize: 12, fontWeight: theme.typography.fontWeightBold }}
      >
        Copyright Â© 2023. All rights reserved
      </Typography>
    </Box>
  );
};
