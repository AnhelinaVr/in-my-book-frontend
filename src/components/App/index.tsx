import { Box, useTheme } from '@mui/material';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import { Alert } from '../Alert';
import { Footer } from '../Footer';
import { Header } from '../Header';
import Home from '../../pages/Home';
import { Paths } from '../../constants';
import { getStyles } from './styles';

const links = [
  { path: Paths.home, name: 'Home' },
  { path: Paths.home, name: 'Rooms' },
  { path: Paths.home, name: 'Books' },
];

const App = () => {
  const theme = useTheme();
  const styles = getStyles(theme);

  return (
    <Box sx={styles.container}>
      <Alert />
      <BrowserRouter>
        <Header isAuthenticated={false} links={links} />
        <Routes>
          <Route path={Paths.home} element={<Home />} />
        </Routes>
        <Footer />
      </BrowserRouter>
    </Box>
  );
};

export default App;
