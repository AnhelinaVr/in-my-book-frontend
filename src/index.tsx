import './index.css';
import App from './components/App';
import { Provider } from 'react-redux';
import React from 'react';
import ReactDOM from 'react-dom/client';
import { ThemeProvider } from '@mui/material';
import store from './redux/store';
import { theme } from './theme/index';

const root = ReactDOM.createRoot(
  document.getElementById('root') as HTMLElement,
);

root.render(
  <React.StrictMode>
    <ThemeProvider theme={theme}>
      <Provider store={store}>
        <App />
      </Provider>
    </ThemeProvider>
  </React.StrictMode>,
);
