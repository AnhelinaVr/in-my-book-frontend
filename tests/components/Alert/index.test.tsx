import { Alert } from '../../../src/components/Alert';
import { AlertType } from '../../../src/types';
import { AppState } from '../../../src/redux/rootReducer';
import React from 'react';
import { fixtures } from './fixtures';
import { render } from '../../helpers/wrapper';
import { screen } from '@testing-library/react';

describe('<Alert />', () => {
  it('should match snapshot', () => {
    const { container } = render(<Alert />, {
      reduxState: fixtures.state,
    });

    expect(container).toMatchSnapshot();
  });

  it('should show error alert', () => {
    render(<Alert />, {
      reduxState: {
        ...fixtures.state,
        alert: {
          ...fixtures.state.alert,
          type: AlertType.ERROR,
        },
      },
    });

    expect(screen.getByTestId(`alert-${AlertType.ERROR}`)).toBeInTheDocument();
    expect(
      screen.queryByTestId(`alert-${AlertType.SUCCESS}`),
    ).not.toBeInTheDocument();
  });

  it('should not show alert if state is empty', () => {
    render(<Alert />, {
      reduxState: {} as AppState,
    });

    expect(screen.queryByTestId(/alert-/)).not.toBeInTheDocument();
  });
});
