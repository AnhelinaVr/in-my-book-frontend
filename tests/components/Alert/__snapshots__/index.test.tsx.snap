// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> should match snapshot 1`] = `
<div>
  <div
    class="MuiSnackbar-root MuiSnackbar-anchorOriginTopCenter css-zzms1-MuiSnackbar-root"
    data-testid="alert-success"
    role="presentation"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardSuccess MuiAlert-standard css-1u4mj23-MuiPaper-root-MuiAlert-root"
      direction="down"
      role="alert"
      style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-12f8ttk-MuiSvgIcon-root"
          data-testid="CheckRoundedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M9 16.17 5.53 12.7a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l4.18 4.18c.39.39 1.02.39 1.41 0L20.29 7.71c.39-.39.39-1.02 0-1.41a.9959.9959 0 0 0-1.41 0L9 16.17z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
      >
        Test message
      </div>
      <div
        class="MuiAlert-action css-ki1hdl-MuiAlert-action"
      >
        <button
          aria-label="Close"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-1e0d89p-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          title="Close"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
